api/v1/submissions/PKPSubmissionFileHandler.inc.php

```diff
@@ -196,23 +198,22 @@ public function getMany($slimRequest, $response, $args)
             }
         }
 
-        \HookRegistry::call('API::submissions::files::params', [&$params, $slimRequest]);
+        $collector = Repo::submissionFiles()
+            ->getCollector()
+            ->filterBySubmissionIds(
+                [$submission->getId()]
+            )
+            ->filterByReviewRoundIds($allowedReviewRoundIds)
+            ->filterByFileStages($allowedFileStages);
 
-        $params['submissionIds'] = [$this->getAuthorizedContextObject(ASSOC_TYPE_SUBMISSION)->getId()];
+        $files = Repo::submissionFiles()->getMany($collector);
 
-        $items = [];
-        $filesIterator = Services::get('submissionFile')->getMany($params);
-        $propertyArgs = [
-            'request' => $request,
-            'slimRequest' => $slimRequest,
-            'submission' => $this->getAuthorizedContextObject(ASSOC_TYPE_SUBMISSION),
-        ];
-        foreach ($filesIterator as $file) {
-            $items[] = Services::get('submissionFile')->getSummaryProperties($file, $propertyArgs);
-        }
+        $items = Repo::submissionFiles()
+            ->getSchemaMap()
+            ->summarizeMany($files);
 
         $data = [
-            'itemsMax' => Services::get('submissionFile')->getCount($params),
+            'itemsMax' => Repo::submissionFiles()->getCount($collector),
```

Comment:
I think `getCount` will re-run the query; I believe you should be able to just use `files->count()`, since it doesn't have a limit applied.